CMSgov/mint-app
# [EASI-2059] CSV Exporting

# EASI-2059

## Changes and Description

- New GraphQL query for retrieving all available data for all models
- Install json2csv package to facilitate front-end csv exporting
- New CsvExportLink utility that encapsulates exporting behavior



## How to test this change

Seed database, click the "Download all plans as CSV button", verify csv against database **paying close attention to embedded structs and arrays** such as:

basics.cmmiGroups, discussions, discussions.replies, etc.


We utilized some key options with json2csv (https://www.npmjs.com/package/json2csv) - the main two being unwind and blankout. These two options are combined to display embedded structs and arrays in the cleanest way possible but they do make the CSV have a considerable amount of whitespace and empty cells.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
